chain/client/src/client.rs

```diff
@@ -1352,8 +1352,13 @@ impl Client {
             self.me.as_ref(),
             self.runtime_adapter.as_ref(),
         )?;
-        persist_chunk(partial_chunk, Some(shard_chunk), self.chain.mut_store())?;
-        self.shards_mgr.distribute_encoded_chunk(encoded_chunk, &merkle_paths, receipts)?;
+        persist_chunk(partial_chunk.clone(), Some(shard_chunk), self.chain.mut_store())?;
+        self.shards_mgr.distribute_encoded_chunk(
+            partial_chunk,
+            encoded_chunk,
+            &merkle_paths,
+            receipts,
+        )?;
         Ok(())
     }
 
```

Comment:
Can we add another CI test (or add an assertion in some existing tests) that covers this failure? It's a bit suboptimal that this is discovered in a nayduck test that is not super relevant to the chunks logic.